server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sClient.java

```diff
@@ -23,26 +23,37 @@
 import io.kubernetes.client.openapi.apis.BatchV1Api;
 import io.kubernetes.client.openapi.apis.CoreV1Api;
 import io.kubernetes.client.openapi.models.*;
+import io.kubernetes.client.util.ClientBuilder;
 import io.kubernetes.client.util.Config;
+import io.kubernetes.client.util.KubeConfig;
 import io.kubernetes.client.util.Yaml;
 import io.kubernetes.client.util.labels.LabelSelector;
+import java.io.FileReader;
 import org.bouncycastle.util.Strings;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
 
-public class Kubernetes {
+@Component
+public class K8sClient {
     private ApiClient client;
     private CoreV1Api coreV1Api;
     private final BatchV1Api batchV1Api;
+
     private final String ns;
 
     /**
      * Basic constructor for Kubernetes
      */
-    public Kubernetes(String ns) throws IOException {
-        client = Config.defaultClient();
+    public K8sClient(@Value("${sw.infra.k8s.name-space}") String ns) throws IOException {
+        String kubeConfigPath = System.getenv("HOME") + "/.kube/config";
+
+        // loading the out-of-cluster config, a kubeconfig from file-system
+        client =
+            ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(new FileReader(kubeConfigPath))).build();
```

Comment:
`Config.defaultClient()`  will do the right actions,  ref: https://github.com/kubernetes-client/java/blob/f8a0654f355878ee0325227c28ca43968933db1e/util/src/main/java/io/kubernetes/client/util/ClientBuilder.java#L103

* KUBECONFIG env
* $HOME/.kube/config
* auto mounted ca
* etc

Reply:
done